project/src/store/offer-data/offer-data.ts

```diff
@@ -33,6 +40,14 @@ const offerData = createReducer(initialState, (builder) => {
     .addCase(loadFavoriteOffers, (state, action) => {
       const { favoriteOffers } = action.payload;
       state.favoriteOffers = favoriteOffers;
+    })
+    .addCase(changeOferFavoriteStatus, (state, action) => {
+      const { changedOffer } = action.payload;
+      state.offers.map((offer) => offer.id === changedOffer.id ? offer.isFavorite = changedOffer.isFavorite : offer.isFavorite);
```

Comment:
map должен что-то возвращать и куда-то присваиваться
```suggestion
      state.offers = state.offers.map((offer) => offer.id === changedOffer.id ? changedOffer : offer);
```
никаких мутаций внутри мапа